### PR TITLE
 Add AWS Integration

### PR DESCRIPTION
**Pull request by MattHodge**
> ⚠️   I am a new Go developer so lots of feedback is wanted! Be critical, please 🤓 

### PR Details

* Add feature to manage the AWS integration with Datadog (closing feature request #159).
* Add acceptence test
* Update **Testify** to use the [ElementsMatch](https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch) in acceptance test

### Notes:
* There isn't a way listed in the [Datadog API](https://docs.datadoghq.com/api/?lang=bash#aws) docs to update AWS Accounts added to the AWS integration so there is no `Update` function.

### Questions:
* Adding an account returns a response like `{'external_id': '123456789'}`. I am currently marshalling this to the `IntegrationAWSAccountCreateResponse` struct. It appears `external_id` is never needed again in further API calls so perhaps I should just return an `error` from the `CreateIntegrationAWS` function and delete this struct?
* Should I add a `GetIntegrationAWSByAccount` or similar function to prevent a consumer of the client having to loop through the results of the `GetIntegrationAWS` function?

### CODE REVIEW

**Comment by iancward** [comment]
The code in `integrations.go` looks pretty much exactly like what I had (I was currently working on the test bits).
Regarding the external_id response to the integration create, I'd leave this in place. My intent with opening the issue for this was so that I could add it to the terraform-datadog provider, which would expose the external_id as an attribute to the resource. It could then be consumed by some other code that created the roles on the AWS side.

I did experiment to see if it was possible to perform a GET request for a specific AWS integration; however, it looks like that's not exposed.

**Comment by iancward** [comment]
I opened a support ticket to Datadog and confirmed that the API for the AWS integration does not yet support a PUT update action. So we're good on that.

**Comment by MattHodge** [comment]
@yfronto Can you provide some feedback on this PR?

File: integrations.go [hunk 1]
```diff
@@ -120,3 +120,65 @@ func (client *Client) GetIntegrationSlack() (*IntegrationSlackRequest, error) {
...
+// GetIntegrationAWS gets all the AWS Accounts in the AWS Integrations from Datadog.
+func (client *Client) GetIntegrationAWS() (*[]IntegrationAWSAccount, error) {
+	var response IntegrationAWSAccountGetResponse
+	if err := client.doJsonRequest("GET", "/v1/integration/aws", nil, &response); err != nil {
+		return nil, err
+	}
+
+	var awsIntegrations []IntegrationAWSAccount
+
+	for _, account := range response.Accounts {
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply integrations.go]: AFAICT
line 174 until 178 can be replaced with:

```return &response.Accounts, nil```

[the remark this answers is not in the document]
Comment (MattHodge) [reply integrations.go]: This has been updated @ojongerius 👍

**Comment by ojongerius** [comment]
Released in [v2.11.0](https://github.com/zorkian/go-datadog-api/releases/tag/v2.11.0)

**Pull request merged**